nova_vm/src/ecmascript/builtins/text_processing/string_objects/string_prototype.rs

```diff
@@ -645,8 +645,85 @@ impl StringPrototype {
         ))
     }
 
-    fn replace(_agent: &mut Agent, _this_value: Value, _: ArgumentsList) -> JsResult<Value> {
-        todo!()
+    /// ### [22.1.3.19 String.prototype.replace ( searchValue, replaceValue )](https://tc39.es/ecma262/multipage/text-processing.html#sec-string.prototype.replace)
+    fn replace(agent: &mut Agent, this_value: Value, args: ArgumentsList) -> JsResult<Value> {
+        // 1. Let O be ? RequireObjectCoercible(this value).
+        let o = require_object_coercible(agent, this_value)?;
+
+        let search_value = args.get(0);
+        let replace_value = args.get(1);
+
+        // 2. If searchValue is neither undefined nor null, then
+        if !search_value.is_null() && !search_value.is_undefined() {
+            let symbol = WellKnownSymbolIndexes::Replace.into();
+            // If replacer is not undefined, Return ? Call(replacer, searchValue, « O, replaceValue »).
+            if let Ok(Some(replacer)) = get_method(agent, search_value, symbol) {
+                return call_function(
+                    agent,
+                    replacer,
+                    search_value,
+                    Some(ArgumentsList(&[o, replace_value])),
+                );
+            }
+        }
+
+        // 3. Let s be ? ToString(O).
+        let s = to_string(agent, o)?;
+
+        // 4. Let searchString be ? ToString(searchValue).
+        let search_string = to_string(agent, search_value)?;
+
+        // 5. Let functionalReplace be IsCallable(replaceValue).
+        let functional_replace = is_callable(replace_value);
+
+        // 6. If functionalReplace is false, Set replaceValue to ? ToString(replaceValue).
+        let replace_value = match functional_replace {
+            None => Some(to_string(agent, replace_value)?),
+            Some(_) => None,
+        };
+
+        // 7. Let searchLength be the length of searchString.
+        let search_length = search_string.len(agent);
+
+        // 8. Let position be StringIndexOf(s, searchString, 0).
+        // 9. If position is not-found, return s.
+        let position = if let Some(position) = search_string.as_str(agent).find(s.as_str(agent)) {
+            position
+        } else {
+            return Ok(s.into_value());
+        };
+
+        // Use rust's replace if the replace is a string, not a function
+        if let Some(replace_value) = replace_value {
```

Comment:
**isse:** Move the `position` calculation from above to below this if check.

Reply:
And actually, that means that you can move this entirely into the None path of the `match functional_replace` statement.